src/custom/conversions.rs

```diff
@@ -0,0 +1,463 @@
+use super::{CodegenExtension, CodegenExtsMap};
+
+use anyhow::{anyhow, Result};
+
+use hugr::{
+    extension::{
+        prelude::{sum_with_error, ConstError},
+        simple_op::MakeExtensionOp,
+        ExtensionId,
+    },
+    ops::{constant::Value, custom::ExtensionOp},
+    std_extensions::arithmetic::{
+        conversions::{self, ConvertOpDef, ConvertOpType},
+        int_types::INT_TYPES,
+    },
+    types::{CustomType, TypeArg},
+    HugrView,
+};
+
+use inkwell::{
+    intrinsics::Intrinsic,
+    types::{BasicType, BasicTypeEnum},
+    values::{AnyValue, BasicValue},
+    FloatPredicate,
+};
+
+use crate::{
+    emit::{
+        func::EmitFuncContext,
+        ops::{emit_custom_unary_op, emit_value},
+        EmitOp, EmitOpArgs,
+    },
+    types::TypingSession,
+};
+
+struct ConversionsEmitter<'c, 'd, H>(&'d mut EmitFuncContext<'c, H>);
+
+impl<'c, H: HugrView> ConversionsEmitter<'c, '_, H> {
+    fn build_trunc_op(
+        &mut self,
+        signed: bool,
+        log_width: u64,
+        args: EmitOpArgs<'c, ExtensionOp, H>,
+    ) -> Result<()> {
+        // Note: This logic is copied from `llvm_type` in the IntTypes
+        // extension. We need to have a common source of truth for this.
+        let (width, int_max_value_s, int_max_value_u) = match log_width {
+            0..=3 => Ok((8, i8::MAX as u64, u8::MAX as u64)),
+            4 => Ok((16, i16::MAX as u64, u16::MAX as u64)),
+            5 => Ok((32, i32::MAX as u64, u32::MAX as u64)),
+            6 => Ok((64, i64::MAX as u64, u64::MAX)),
+            m => Err(anyhow!(
```

Comment:
nit: if you put `return` before the `Err` you can avoid all 4 `Ok`'s and the trailing `?`